In [75]:
print("CodeAlpha EDA Project Started Successfully!")



CodeAlpha EDA Project Started Successfully!


In [76]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("All libraries imported successfully!")

All libraries imported successfully!


In [77]:
df = pd.read_csv("dataset/ecommerce_sales_analysis.csv")

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [78]:
df.head()


,product_id,product_name,category,price,review_score,review_count,sales_month_1,sales_month_2,sales_month_3,sales_month_4,sales_month_5,sales_month_6,sales_month_7,sales_month_8,sales_month_9,sales_month_10,sales_month_11,sales_month_12
0,1,Product_1,Clothing,190.40,1.7,220,479,449,92,784,604,904,446,603,807,252,695,306
1,2,Product_2,Home & Kitchen,475.60,3.2,903,21,989,861,863,524,128,610,436,176,294,772,353
2,3,Product_3,Toys,367.34,4.5,163,348,558,567,143,771,409,290,828,340,667,267,392
3,4,Product_4,Toys,301.34,3.9,951,725,678,59,15,937,421,670,933,56,157,168,203
4,5,Product_5,Books,82.23,4.2,220,682,451,649,301,620,293,411,258,854,548,770,257


In [79]:
df.shape


(1000, 18)

In [80]:
df.columns


Index(['product_id', 'product_name', 'category', 'price', 'review_score',
       'review_count', 'sales_month_1', 'sales_month_2', 'sales_month_3',
       'sales_month_4', 'sales_month_5', 'sales_month_6', 'sales_month_7',
       'sales_month_8', 'sales_month_9', 'sales_month_10', 'sales_month_11',
       'sales_month_12'],
      dtype='object')

In [81]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 18 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   product_id      1000 non-null   int64  
 1   product_name    1000 non-null   object 
 2   category        1000 non-null   object 
 3   price           1000 non-null   float64
 4   review_score    1000 non-null   float64
 5   review_count    1000 non-null   int64  
 6   sales_month_1   1000 non-null   int64  
 7   sales_month_2   1000 non-null   int64  
 8   sales_month_3   1000 non-null   int64  
 9   sales_month_4   1000 non-null   int64  
 10  sales_month_5   1000 non-null   int64  
 11  sales_month_6   1000 non-null   int64  
 12  sales_month_7   1000 non-null   int64  
 13  sales_month_8   1000 non-null   int64  
 14  sales_month_9   1000 non-null   int64  
 15  sales_month_10  1000 non-null   int64  
 16  sales_month_11  1000 non-null   int64  
 17  sales_month_12  1000 non-null   in

In [82]:
print("Missing values in each column:")
print(df.isnull().sum())

Missing values in each column:
product_id        0
product_name      0
category          0
price             0
review_score      0
review_count      0
sales_month_1     0
sales_month_2     0
sales_month_3     0
sales_month_4     0
sales_month_5     0
sales_month_6     0
sales_month_7     0
sales_month_8     0
sales_month_9     0
sales_month_10    0
sales_month_11    0
sales_month_12    0
dtype: int64


In [83]:
print("Total missing values:", df.isnull().sum().sum())

Total missing values: 0


In [84]:
print("Number of duplicate rows:", df.duplicated().sum())

Number of duplicate rows: 0


In [85]:
print("Product categories:")
print(df["category"].unique())


Product categories:
['Clothing' 'Home & Kitchen' 'Toys' 'Books' 'Electronics' 'Health'
 'Sports']


In [86]:
category_counts = df["category"].value_counts()

print(category_counts)

category
Books             154
Sports            153
Toys              151
Clothing          140
Health            139
Electronics       138
Home & Kitchen    125
Name: count, dtype: int64


In [87]:
sales_columns = [
    f"sales_month_{i}" for i in range(1, 13)
]

In [88]:
df["total_sales"] = df[sales_columns].sum(axis=1)

df[["product_name", "category", "total_sales"]].head()

,product_name,category,total_sales
0,Product_1,Clothing,6421
1,Product_2,Home & Kitchen,6027
2,Product_3,Toys,5580
3,Product_4,Toys,5022
4,Product_5,Books,6094


In [89]:
df["estimated_revenue"] = df["price"] * df["total_sales"]

df[
    [
        "product_name",
        "price",
        "total_sales",
        "estimated_revenue"
    ]
].head()

,product_name,price,total_sales,estimated_revenue
0,Product_1,190.40,6421,1222558.40
1,Product_2,475.60,6027,2866441.20
2,Product_3,367.34,5580,2049757.20
3,Product_4,301.34,5022,1513329.48
4,Product_5,82.23,6094,501109.62


In [90]:
print("Total number of products:", df["product_id"].nunique())

print("Total units sold:", df["total_sales"].sum())

print("Average product price: $", round(df["price"].mean(), 2))

print("Average review score:", round(df["review_score"].mean(), 2))

print(
    "Total estimated revenue: $",
    round(df["estimated_revenue"].sum(), 2)
)

Total number of products: 1000
Total units sold: 6019912
Average product price: $ 247.68
Average review score: 3.03
Total estimated revenue: $ 1488704183.02


In [91]:
top_products = df.nlargest(
    10,
    "total_sales"
)[
    [
        "product_name",
        "category",
        "total_sales"
    ]
]

top_products

,product_name,category,total_sales
223,Product_224,Electronics,9151
285,Product_286,Clothing,8921
733,Product_734,Health,8914
904,Product_905,Sports,8783
179,Product_180,Sports,8775
852,Product_853,Books,8765
238,Product_239,Health,8724
923,Product_924,Electronics,8525
936,Product_937,Electronics,8459
196,Product_197,Toys,8418


In [92]:
top_revenue_products = df.nlargest(
    10,
    "estimated_revenue"
)[
    [
        "product_name",
        "category",
        "estimated_revenue"
    ]
]


top_revenue_products

,product_name,category,estimated_revenue
305,Product_306,Books,3812158.55
531,Product_532,Books,3763945.80
52,Product_53,Sports,3722796.00
228,Product_229,Electronics,3698409.92
390,Product_391,Books,3650361.00
522,Product_523,Toys,3566043.61
140,Product_141,Books,3549917.34
112,Product_113,Clothing,3520168.40
751,Product_752,Health,3519126.48
475,Product_476,Toys,3515858.12


In [93]:
category_sales = (
    df.groupby("category")["total_sales"]
    .sum()
    .sort_values(ascending=False)
)

category_sales

category
Books             938229
Toys              917101
Sports            916371
Electronics       845120
Health            834414
Clothing          826536
Home & Kitchen    742141
Name: total_sales, dtype: int64

In [94]:
category_revenue = (
    df.groupby("category")["estimated_revenue"]
    .sum()
    .sort_values(ascending=False)
)

category_revenue

category
Books             2.367828e+08
Sports            2.326488e+08
Toys              2.302372e+08
Health            2.217369e+08
Electronics       2.016747e+08
Clothing          1.872583e+08
Home & Kitchen    1.783655e+08
Name: estimated_revenue, dtype: float64

In [95]:
category_reviews = (
    df.groupby("category")["review_score"]
    .mean()
    .sort_values(ascending=False)
)

category_reviews

category
Electronics       3.143478
Books             3.096753
Sports            3.091503
Home & Kitchen    3.035200
Health            3.005036
Clothing          2.952143
Toys              2.870861
Name: review_score, dtype: float64

In [96]:
monthly_sales = df[sales_columns].sum()

monthly_sales

sales_month_1     498306
sales_month_2     507661
sales_month_3     506739
sales_month_4     503823
sales_month_5     487194
sales_month_6     491653
sales_month_7     507011
sales_month_8     504569
sales_month_9     491934
sales_month_10    514798
sales_month_11    505838
sales_month_12    500386
dtype: int64

In [97]:
monthly_sales.index = [
    f"Month {i}" for i in range(1, 13)
]

monthly_sales

Month 1     498306
Month 2     507661
Month 3     506739
Month 4     503823
Month 5     487194
Month 6     491653
Month 7     507011
Month 8     504569
Month 9     491934
Month 10    514798
Month 11    505838
Month 12    500386
dtype: int64

In [98]:
print("Highest sales month:", monthly_sales.idxmax())
print("Units sold:", monthly_sales.max())

Highest sales month: Month 10
Units sold: 514798


In [99]:
print("Lowest sales month:", monthly_sales.idxmin())
print("Units sold:", monthly_sales.min())

Lowest sales month: Month 5
Units sold: 487194


In [100]:
correlation_price_sales = df["price"].corr(df["total_sales"])

print(
    "Correlation between price and total sales:",
    round(correlation_price_sales, 3)
)

Correlation between price and total sales: -0.016


In [101]:
correlation_reviews_sales = df["review_score"].corr(
    df["total_sales"]
)

print(
    "Correlation between review score and total sales:",
    round(correlation_reviews_sales, 3)
)

Correlation between review score and total sales: -0.018


In [102]:
correlation_review_count_sales = df["review_count"].corr(
    df["total_sales"]
)

print(
    "Correlation between review count and total sales:",
    round(correlation_review_count_sales, 3)
)

Correlation between review count and total sales: -0.069


In [103]:
highest_rated = df.sort_values(
    by=["review_score", "review_count"],
    ascending=[False, False]
)[
    [
        "product_name",
        "category",
        "review_score",
        "review_count"
    ]
].head(10)

highest_rated

,product_name,category,review_score,review_count
11,Product_12,Health,5.0,991
333,Product_334,Health,5.0,935
715,Product_716,Sports,5.0,921
348,Product_349,Clothing,5.0,815
591,Product_592,Clothing,5.0,777
664,Product_665,Books,5.0,755
584,Product_585,Toys,5.0,704
74,Product_75,Toys,5.0,629
889,Product_890,Home & Kitchen,5.0,435
464,Product_465,Clothing,5.0,432


In [104]:
Q1 = df["price"].quantile(0.25)
Q3 = df["price"].quantile(0.75)

IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

price_outliers = df[
    (df["price"] < lower_bound) |
    (df["price"] > upper_bound)
]

print("Number of price outliers:", len(price_outliers))

Number of price outliers: 0


In [105]:
Q1 = df["total_sales"].quantile(0.25)
Q3 = df["total_sales"].quantile(0.75)

IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

sales_outliers = df[
    (df["total_sales"] < lower_bound) |
    (df["total_sales"] > upper_bound)
]

print("Number of total sales outliers:", len(sales_outliers))

Number of total sales outliers: 13


In [106]:
correlation_columns = [
    "price",
    "review_score",
    "review_count",
    "total_sales",
    "estimated_revenue"
]

correlation_matrix = df[correlation_columns].corr()

correlation_matrix

,price,review_score,review_count,total_sales,estimated_revenue
price,1.000000,0.028960,0.042189,-0.015978,0.950023
review_score,0.028960,1.000000,0.027351,-0.018186,0.027932
review_count,0.042189,0.027351,1.000000,-0.069393,0.026334
total_sales,-0.015978,-0.018186,-0.069393,1.000000,0.256792
estimated_revenue,0.950023,0.027932,0.026334,0.256792,1.000000


In [107]:
df.to_csv(
    "dataset/ecommerce_sales_cleaned.csv",
    index=False
)

print("Cleaned dataset saved successfully!")

Cleaned dataset saved successfully!


In [108]:
print("===== KEY INSIGHTS DATA =====")

# 1. Dataset information
print("\n1. DATASET OVERVIEW")
print("Total products:", df["product_id"].nunique())
print("Total categories:", df["category"].nunique())

# 2. Best-selling category
category_sales = df.groupby("category")["total_sales"].sum().sort_values(ascending=False)

print("\n2. CATEGORY PERFORMANCE")
print("Best-selling category:", category_sales.idxmax())
print("Total units sold:", category_sales.max())
print("Lowest-selling category:", category_sales.idxmin())
print("Total units sold:", category_sales.min())

# 3. Top-selling product
top_product = df.loc[df["total_sales"].idxmax()]

print("\n3. TOP-SELLING PRODUCT")
print("Product:", top_product["product_name"])
print("Category:", top_product["category"])
print("Total units sold:", top_product["total_sales"])

# 4. Monthly sales performance
monthly_sales = df[sales_columns].sum()
monthly_sales.index = [f"Month {i}" for i in range(1, 13)]

print("\n4. MONTHLY SALES")
print("Highest sales month:", monthly_sales.idxmax())
print("Units sold:", monthly_sales.max())
print("Lowest sales month:", monthly_sales.idxmin())
print("Units sold:", monthly_sales.min())

# 5. Correlations
price_sales_corr = df["price"].corr(df["total_sales"])
rating_sales_corr = df["review_score"].corr(df["total_sales"])
review_count_sales_corr = df["review_count"].corr(df["total_sales"])

print("\n5. CORRELATIONS")
print("Price vs Total Sales:", round(price_sales_corr, 3))
print("Review Score vs Total Sales:", round(rating_sales_corr, 3))
print("Review Count vs Total Sales:", round(review_count_sales_corr, 3))

# 6. Outliers
print("\n6. OUTLIERS")
print("Number of price outliers:", len(price_outliers))
print("Number of sales outliers:", len(sales_outliers))

# 7. Estimated revenue
print("\n7. ESTIMATED REVENUE")
print("Total estimated annual revenue: $", round(df["estimated_revenue"].sum(), 2))

===== KEY INSIGHTS DATA =====

1. DATASET OVERVIEW
Total products: 1000
Total categories: 7

2. CATEGORY PERFORMANCE
Best-selling category: Books
Total units sold: 938229
Lowest-selling category: Home & Kitchen
Total units sold: 742141

3. TOP-SELLING PRODUCT
Product: Product_224
Category: Electronics
Total units sold: 9151

4. MONTHLY SALES
Highest sales month: Month 10
Units sold: 514798
Lowest sales month: Month 5
Units sold: 487194

5. CORRELATIONS
Price vs Total Sales: -0.016
Review Score vs Total Sales: -0.018
Review Count vs Total Sales: -0.069

6. OUTLIERS
Number of price outliers: 0
Number of sales outliers: 13

7. ESTIMATED REVENUE
Total estimated annual revenue: $ 1488704183.02


## Key Insights

1. **Dataset Overview:** The dataset contains 1,000 unique e-commerce products distributed across 7 different product categories.

2. **Category Performance:** Books was the best-selling product category with a total of 938,229 units sold, while Home & Kitchen recorded the lowest sales with 742,141 units sold.

3. **Top-Selling Product:** Product_224 from the Electronics category was the highest-selling individual product, recording 9,151 units sold across the 12-month period.

4. **Monthly Sales Trends:** Month 10 recorded the highest overall sales with 514,798 units sold, whereas Month 5 had the lowest sales with 487,194 units sold.

5. **Price and Sales Relationship:** The correlation between product price and total sales was -0.016, indicating virtually no linear relationship between product price and sales volume in this dataset.

6. **Outlier Detection:** The IQR method identified 13 potential outliers in total product sales, highlighting products with unusually high or low sales compared with the overall distribution.

7. **Estimated Revenue:** The total estimated annual revenue across all products was approximately $1.49 billion, calculated by multiplying each product's listed price by its total annual unit sales.

## Conclusion

The exploratory data analysis revealed meaningful patterns in product performance, category-level sales, monthly trends, pricing relationships, and potential anomalies. Books emerged as the strongest category by unit sales, while Product_224 was the highest-selling individual product. Sales remained relatively consistent across the 12 months, with Month 10 recording the highest performance. The near-zero correlation between price and total sales suggests that product price alone was not a significant linear predictor of sales volume in this dataset.